WordPress/Classes/Services/PlanService.swift

```diff
@@ -31,92 +37,87 @@ open class PlanService: LocalCoreDataService {
 
         }, failure: failure)
     }
 
-    func mergeRemoteWpcomPlans(_ remotePlans: [RemoteWpcomPlan],
+    private func mergeRemoteWpcomPlans(_ remotePlans: [RemoteWpcomPlan],
                                remoteGroups: [RemotePlanGroup],
                                remoteFeatures: [RemotePlanFeature],
                                onComplete: @escaping () -> Void ) {
-
-        mergeRemoteWpcomPlans(remotePlans)
-        mergeRemotePlanGroups(remoteGroups)
-        mergeRemotePlanFeatures(remoteFeatures)
-
-        ContextManager.sharedInstance().save(managedObjectContext) {
-            onComplete()
-        }
+        coreDataStack.performAndSave({ context in
+            self.mergeRemoteWpcomPlans(remotePlans, in: context)
+            self.mergeRemotePlanGroups(remoteGroups, in: context)
+            self.mergeRemotePlanFeatures(remoteFeatures, in: context)
+        }, completion: onComplete)
     }
 
 
-    func allPlans() -> [Plan] {
+    func allPlans(in context: NSManagedObjectContext) -> [Plan] {
         let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: "Plan")
         fetchRequest.sortDescriptors = [NSSortDescriptor(key: "order", ascending: true)]
         do {
-            return try managedObjectContext.fetch(fetchRequest) as! [Plan]
+            return try context.fetch(fetchRequest) as! [Plan]
         } catch let error as NSError {
             DDLogError("Error fetching Plans: \(error.localizedDescription)")
             return [Plan]()
         }
     }
 
 
-    func findPlanByShortname(_ shortname: String) -> Plan? {
-        let plans = allPlans() as NSArray
+    private func findPlanByShortname(_ shortname: String, in context: NSManagedObjectContext) -> Plan? {
+        let plans = allPlans(in: context) as NSArray
         let results = plans.filtered(using: NSPredicate(format: "shortname = %@", shortname))
         return results.first as? Plan
     }
 
 
-    func allPlanGroups() -> [PlanGroup] {
+    private func allPlanGroups(in context: NSManagedObjectContext) -> [PlanGroup] {
         let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: "PlanGroup")
         fetchRequest.sortDescriptors = [NSSortDescriptor(key: "order", ascending: true)]
         do {
-            return try managedObjectContext.fetch(fetchRequest) as! [PlanGroup]
+            return try context.fetch(fetchRequest) as! [PlanGroup]
         } catch let error as NSError {
             DDLogError("Error fetching PlanGroups: \(error.localizedDescription)")
             return [PlanGroup]()
         }
     }
 
 
-    func findPlanGroupBySlug(_ slug: String) -> PlanGroup? {
-        let groups = allPlanGroups() as NSArray
+    private func findPlanGroupBySlug(_ slug: String, in context: NSManagedObjectContext) -> PlanGroup? {
+        let groups = allPlanGroups(in: context) as NSArray
         let results = groups.filtered(using: NSPredicate(format: "slug = %@", slug))
         return results.first as? PlanGroup
```

Comment:
Similar question and suggestion as for `findPlanByShortname`:

```suggestion
        return allPlanGroups(in: context).filter { $0.slug == slug }.first
```

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/19949/commits/5ba955d546f834cc99ac058b2f9d0f4661cb1f45